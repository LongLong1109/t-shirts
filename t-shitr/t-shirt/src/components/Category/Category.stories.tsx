import type { Meta, StoryObj } from '@storybook/react'

import Category from '.'

const meta = {
  title: 'Components/Category',
  component: Category,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof Category>

export default meta
type Story = StoryObj<typeof Category>

export const CategoryPrimary: Story = {
  args: {},
}
