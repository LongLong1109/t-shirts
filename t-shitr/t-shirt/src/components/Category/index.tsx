'use client'
import { useState } from 'react'
import Image from 'next/image'

// components
import Carousel from '@/components/Carousel'

// mock
import { categoriesStyle } from '@/mocks'

const Category = () => {
  const [currentCategory, setCurrentCategory] = useState(0)

  return (
    <div className='flex justify-end items-center gap-4 sm:flex-col md:flex-row'>
      <Carousel
        images={categoriesStyle}
        style='max-w-[360px] max-h-[230px]'
        currentIndex={currentCategory}
        setCurrentIndex={setCurrentCategory}
      />

      <Image
        src={categoriesStyle[(currentCategory + 1) % categoriesStyle.length].src}
        alt={
          categoriesStyle[(currentCategory - 1 + categoriesStyle.length) % categoriesStyle.length]
            .alt
        }
        width={250}
        height={150}
      />
    </div>
  )
}

export default Category
