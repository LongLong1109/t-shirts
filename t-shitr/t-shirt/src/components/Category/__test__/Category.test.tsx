import { render } from '@testing-library/react';

import Category from '..';

describe('Category Component', () => {
  it('should matches snapshot', () => {
    const { asFragment } = render(<Category />);
    expect(asFragment()).toMatchSnapshot();
  });
});
