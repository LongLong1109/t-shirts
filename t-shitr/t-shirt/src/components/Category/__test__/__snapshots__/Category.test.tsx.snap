// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-end items-center gap-4 sm:flex-col md:flex-row"
  >
    <div
      class="relative mx-auto overflow-hidden max-w-[360px] max-h-[230px]"
    >
      <div
        class="flex transition-transform ease-in-out duration-1000"
        style="transform: translateX(-0%);"
      >
        <div
          class="min-w-full flex justify-center gap-8"
        >
          <img
            alt="greenShirts"
            class="w-undefined"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=828&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=828&q=75 2x"
            style="color: transparent;"
            width="400"
          />
        </div>
        <div
          class="min-w-full flex justify-center gap-8"
        >
          <img
            alt="jean"
            class="w-undefined"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=828&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=828&q=75 2x"
            style="color: transparent;"
            width="400"
          />
        </div>
      </div>
      <button
        class="absolute top-1/2 transform -translate-y-1/2 bg-blue-500 text-white-100 p-2 z-40"
        data-testid="left-btn"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.6201 4.49999L8.50012 11.62L15.6201 18.74"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-miterlimit="10"
            stroke-width="2"
          />
        </svg>
      </button>
      <button
        class="absolute top-1/2 right-0 transform -translate-y-1/2 bg-blue-500 text-white-100 p-2"
        data-testid="right-btn"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.37988 4.49999L15.4999 11.62L8.37988 18.74"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-miterlimit="10"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <img
      alt="jean"
      data-nimg="1"
      decoding="async"
      height="150"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
      style="color: transparent;"
      width="250"
    />
  </div>
</DocumentFragment>
`;
